### PR TITLE
Float specs

### PR DESCRIPTION
**Pull request by chrisseaton**
@eregon we weren't certain about this - does it make sense to have `inspect` and `to_s` factored out but both tested like this?

### CODE REVIEW

File: spec/ruby/core/float/negative_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    it "returns false" do
+      0.1.negative?.should be_false
+    end
+  end
+
+  describe "on zero" do
+    it "returns false" do
+      0.0.negative?.should be_false
+    end
+  end
```
Comment (eregon) [inline spec/ruby/core/float/negative_spec.rb:14]: Would be interesting to test on `-0.0` too

File: spec/ruby/core/float/positive_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    it "returns true" do
+      0.1.positive?.should be_true
+    end
+  end
+
+  describe "on zero" do
+    it "returns false" do
+      0.0.positive?.should be_false
+    end
+  end
```
Comment (eregon) [inline spec/ruby/core/float/positive_spec.rb:14]: Similarly, it would be interesting to test on `-0.0` too

Comment (chrisseaton) [reply spec/ruby/core/float/positive_spec.rb:14]: Done.

File: spec/ruby/core/float/negative_spec.rb (hunk 1 above)
Comment (chrisseaton) [reply spec/ruby/core/float/negative_spec.rb:14]: Done... but I think there's a bug in MRI here https://bugs.ruby-lang.org/issues/17098.

File: spec/ruby/core/float/shared/to_s.rb [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+      (4.8 / 8.0).send(@method).should == "0.6"
+      (9.0 / 2.5).send(@method).should == "3.6"
+      (0.2 / 0.6).send(@method).should == "0.33333333333333337"
+      (7.8 / 5.2).send(@method).should == "1.5"
+      (9.5 / 5.5).send(@method).should == "1.7272727272727273"
+    end
+  end
+end
+
+describe :float_to_s, shared: true do
```
Comment (eregon) [inline spec/ruby/core/float/shared/to_s.rb:290]: This overrides the above `describe :float_to_s, shared: true do`, I'll fix it.

Comment (eregon) [reply spec/ruby/core/float/shared/to_s.rb:290]: And https://github.com/ruby/mspec/commit/64c41b670b690bc74dfc5d129eb7a629d6fa8490 to catch this earlier.

**Pull request merged**